File: programming-concepts/iteration/visual-basic/while_average_mark.vb
' Raspberry Pi Foundation
' Developed as part of Ada Computer Science
' Usage licensed under CC BY-NC-SA 4.0
' Note: This file is designed to be copied out and compiled on your machine.
' To run this file you need to:
' 1. Copy the contents
' 2. Paste them into the Visual Basic IDE of your choice (Visual Studio, for example)
' 3. Compile the program
' 4. Run the program

Module Program

    ' The Main subroutine is the default entry point for a VB program
    Sub Main()
        Dim sum As Integer = 0
        Dim numValues As Integer = 0

        Console.WriteLine("Enter a mark or -1 to end ")
        Dim userInput As String = Console.ReadLine()
        Dim mark As Integer = Int32.Parse(userInput)

        While mark <> -1
            sum = sum + mark
            numValues = numValues + 1
            Console.WriteLine("Enter a mark or -1 to end ")
            userInput = Console.ReadLine()
            mark = Int32.Parse(userInput)
        End While

        Dim average As Double = CDbl(sum) / CDbl(numValues)
        Console.WriteLine($"The average mark was {average}")
    End Sub

End Module

File: programming-concepts/iteration/java/while_average_mark.java
/*
Raspberry Pi Foundation
Developed as part of Ada Computer Science

Usage licensed under CC BY-NC-SA 4.0

Note: This file is designed to be copied out and compiled on your machine.
In order for the program to compile properly in some IDEs, you need to ensure that the
class filename is the same as the class name with the main method in it.

To run this file you need to:
1. Copy the contents
2. Create a Java project in the IDE of your choice (Eclipse, for example)
3. Create a new Java Class that uses the same name as the class containing the main method in this program
4. Paste the contents into the new Java Class
5. Save and run the program
*/

import java.util.Scanner;

class Iteration {
    // The main method is the entry point for all Java programs
    public static void main(String[] args) {
        // Use Scanner for reading user input from the console
        Scanner scanner = new Scanner(System.in);

        int sum = 0;
        int numValues = 0;

        System.out.println("Enter a mark or -1 to end ");
        String userInput = scanner.nextLine();
        int mark = Integer.parseInt(userInput);

        while (mark != -1) {
            sum = sum + mark;
            numValues = numValues + 1;

            System.out.println("Enter a mark or -1 to end ");
            userInput = scanner.nextLine();
            mark = Integer.parseInt(userInput);
        }

        double average = (double)sum / (double)numValues;
        System.out.println("The average mark was " + average);
    }


}